### ISSUE TITLE
Feature request: Don't decorate enums with a [Flag] attribute that based on char values

### ISSUE THREAD

**Issue by Corniel** [SonarSource/sonar-dotnet#2396]
### Description
C# allows you (VB.NET does not seems to support this at all), allow you to link enum values to `char` literals. That might be okay in itself, but combining that with a [`Flags`] attribute is highly confusing, an far besides the point to use `char`s to begin with to define the value of the enum fields. Therefor, I suggest a rule that discourages this practice.

### Expected behavior
``` C#
[Flags]
public enum CharBasedEnumsShouldNotBeDecoratedWithFlags // noncompliant
{
    A = 'A',
    B = 'B',
    C = 'C',
    ABC = A | B | C,
}
```

**Comment by andrei-epure-sonarsource** [comment]
Thanks @Corniel ! I finally got to look at this.
Your code sample already triggers [RSPEC-4070](https://rules.sonarsource.com/csharp/RSPEC-4070). I will close this as a duplicate, please feel free to tell me if I am wrong.

**Comment by Corniel** [comment]
@andrei-epure-sonarsource In most cases this will safe the day. However, I assume that if you choose the `char` values that match the right int values, that rule will not be triggered. So whatever you choose, `char` should always be invalid, and documented explicitly as that.
